compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,33 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        @tailrec
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)], 
+            namesSeen: Set[String] = Set.empty): Unit = {
+          pairs match {
+            case (js.StringLiteral(keyName), _) :: tail =>
+              if (namesSeen(keyName)) {
+                reporter.warning(pos, 
+                    s"Duplicate key in object literal: $keyName. Only the last occurrence is assigned.")
+              }
+              warnIfDuplicatedKey(tail, namesSeen + keyName)
+            case Nil => // Do nothing and finish
+          }
```

Comment:
The type of the list is `List[(js.StringLiteral, ...)]`, isn't that enough to ensure that the `unapply` will succeed?